### PR TITLE
Extend agent telemetry with agent CPU/memory usage

### PR DESCRIPTION
**Pull request by thatplguy**

### CODE REVIEW

File: api_schema/schema.go [hunk 1]
```diff
@@ -524,6 +524,17 @@ type PostClientPacketCaptureStatsRequest struct {
...
 	ApidumpError     ApidumpErrorType `json:"apidump_error,omitempty"`
 	ApidumpErrorText string           `json:"apidump_error_text,omitempty"`
+
+	// Report CPU and memory usage, if available.
+	AgentUsage *AgentUsage `json:"agent_usage,omitempty"`
+}
+
+type AgentUsage struct {
+	// Akita agent CPU usage, as a percentage of total CPU availability.
+	RelativeCPU float64 `json:"relative_cpu"`
```
Comment (mgritter) [inline api_schema/schema.go:534]: This does not seem very helpful without context-- how many CPUs are in the container or VM or host?

I suppose the user might know the context already, but it seems confusing to have this be a relative number.

File: api_schema/schema.go [hunk 2]
```diff
@@ -524,6 +524,17 @@ type PostClientPacketCaptureStatsRequest struct {
 	// Report on any errors encounted during apidump that should be shown to the user.
 	ApidumpError     ApidumpErrorType `json:"apidump_error,omitempty"`
 	ApidumpErrorText string           `json:"apidump_error_text,omitempty"`
+
+	// Report CPU and memory usage, if available.
+	AgentUsage *AgentUsage `json:"agent_usage,omitempty"`
```
Comment (liujed) [inline api_schema/schema.go:529]: `AgentResourceUsage` and `agent_resource_usage` would be more descriptive.

File: api_schema/schema.go [hunk 3]
```diff
@@ -524,6 +524,17 @@ type PostClientPacketCaptureStatsRequest struct {
...
+}
+
+type AgentUsage struct {
+	// Akita agent CPU usage, as a percentage of total CPU availability.
+	RelativeCPU float64 `json:"relative_cpu"`
+
+	// Peak virtual memory size in bytes.
+	VMPeak uint64 `json:"vm_peak"`
```
Comment (liujed) [inline api_schema/schema.go:537]: This should be peak RSS.

**Review by liujed — APPROVED** [review]

File: api_schema/schema.go (hunk 2 above)
Comment (thatplguy) [reply api_schema/schema.go]: Changed in 9a50cbd4827ff573e2d59bc69a8cfe7f73971278.

File: api_schema/schema.go (hunk 3 above)
Comment (thatplguy) [reply api_schema/schema.go]: Fixed in 9a50cbd4827ff573e2d59bc69a8cfe7f73971278.

**Pull request merged**